src/container/amlManager.ts

```diff
@@ -0,0 +1,159 @@
+                            });
+                        }),
+                );
+            }
+
+            const workspaceItems: AmlWorkspaceQuickPickItem[] = await Utility.awaitPromiseArray<AmlWorkspaceQuickPickItem>(workspacePromises, Constants.amlWorkspaceDesc);
+            return workspaceItems;
+        } catch (error) {
+            error.message = `Error fetching workspace list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private async selectImage(workspace: Workspace, session: AzureSession): Promise<vscode.QuickPickItem> {
+        const imageItem: vscode.QuickPickItem = await vscode.window.showQuickPick(this.loadAmlImageItems(workspace, session),
+            { placeHolder: "Select Image", ignoreFocusOut: true });
+        return imageItem;
+    }
+
+    private async loadAmlImageItems(workspace: Workspace, session: AzureSession): Promise<vscode.QuickPickItem[]> {
+        try {
+            const modelMgmtEndpoint: string = await this.getModelMgmtEndpoint(workspace);
+            const credentials: ServiceClientCredentials = session.credentials;
+            const client: ServiceClient = new ServiceClient(credentials);
+            const result: HttpOperationResponse<{ value: IImage[] }> = await client.sendRequestWithHttpOperationResponse<{ value: IImage[] }>({
+                method: "GET",
+                baseUrl: modelMgmtEndpoint,
+                pathTemplate: `/api${workspace.id}/images`,
+                queryParameters: { "api-version": Constants.amlApiVersion },
+                serializationMapper: undefined,
+                deserializationMapper: undefined,
+            });
+
+            if (result.body.value === undefined || result.body.value.length === 0) {
+                throw new Error("No image can be found in the workspace.");
+            }
+
+            return result.body.value.map((image: IImage) => {
+                return { label: image.name, description: image.imageLocation };
+            });
+        } catch (error) {
+            error.message = `Error fetching image list: ${error.message}`;
+            throw error;
+        }
+    }
+
+    private getWorkspaceNameCounts(workspaces: Workspace[]): Map<string, number> {
+        const counts: Map<string, number> = new Map<string, number>();
+        workspaces.forEach((workspace: Workspace) => {
+            counts.set(workspace.name, counts.has(workspace.name) ? counts.get(workspace.name) + 1 : 1);
+        });
+
+        return counts;
+    }
+
+    private async getModelMgmtEndpoint(workspace: Workspace): Promise<string> {
+        const historyEndpoint: string = await this.getHistoryEndpoint(workspace);
+        let location: string;
+        const res: string[] = historyEndpoint.match(/\/\/(.*?)\./);
+        if (res.length < 2) {
+            throw new Error("No available endpoint");
+        } else {
+            location = res[1];
+        }
+
+        return `https://${location}.modelmanagement.azureml.net`;
```

Comment:
why not just get the modelmanagement from response instead of calculate base don location? It may cause potential error if it is not global azure

Reply:
Code in this method is copied from AI tools, and I cannot find any documentation about  the REST API of Azure Machine Learning Model Management services. There might be some edge cases where we cannot simply get the model management endpoint from the discovery URL.